src/test/java/com/synopsys/integration/jenkins/wrapper/JenkinsProxyHelperTest.java

```diff
@@ -58,94 +55,83 @@
-        Mockito.when(proxyConfigurationMock.getUserName()).thenReturn(proxyGetUserName);
-        Mockito.when(proxyConfigurationMock.getPassword()).thenReturn(expectedProxyPassword);
-        Mockito.when(proxyConfigurationMock.getNoProxyHostPatterns()).thenReturn(ignoredProxyHosts);
+    public void testFromProxyConfigurationValidProxyConfig(List<Pattern> unusedProxyHosts, String expectedNtlmDomain, String expectedProxyUsername) {
+        ProxyInfo expected = createExpectedProxyInfo(expectedNtlmDomain,null);
+        ProxyConfiguration inputProxyConfiguration = createProxyConfiguration(expectedProxyUsername, expectedProxyPassword);
+        ProxyInfo actual = JenkinsProxyHelper.fromProxyConfiguration(inputProxyConfiguration).getProxyInfo(hostToInclude);
+        assertEquals(expected, actual);
+    }
 
-        ProxyInfo proxyInfo = JenkinsProxyHelper.fromJenkins(jenkinsWrapper).getProxyInfo(url);
-        validProxyInfoValidation(proxyInfo, expectedNtlmDomain, StringUtils.EMPTY);
+    private ProxyInfo createExpectedProxyInfo(String ntlmDomain, String ntlmWorkstation) {
+        ProxyInfoBuilder proxyInfoBuilder = new ProxyInfoBuilder();
+        CredentialsBuilder credentialsBuilder = new CredentialsBuilder();
+        credentialsBuilder.setUsernameAndPassword(expectedProxyUsername, expectedProxyPassword);
+        credentialsBuilder.setPassword(expectedProxyPassword);
+        proxyInfoBuilder.setHost(expectedProxyHost);
+        proxyInfoBuilder.setPort(expectedProxyPort);
+        proxyInfoBuilder.setNtlmDomain(ntlmDomain);
+        proxyInfoBuilder.setNtlmWorkstation(ntlmWorkstation);
+        proxyInfoBuilder.setCredentials(credentialsBuilder.build());
+
+        return proxyInfoBuilder.build();
     }
 
-    private void validProxyInfoValidation(ProxyInfo proxyInfo, String ntlmDomain, String ntlmWorkstation) {
-        assertTrue(proxyInfo.getProxy().isPresent());
-        assertTrue(proxyInfo.getHost().isPresent());
-        assertEquals(expectedProxyHost, proxyInfo.getHost().get());
-        assertEquals(expectedProxyPort, proxyInfo.getPort());
-        assertTrue(proxyInfo.getProxyCredentials().isPresent());
-        assertEquals(expectedProxyUsername, proxyInfo.getUsername().orElse(null));
-        assertEquals(expectedProxyPassword, proxyInfo.getPassword().orElse(null));
-        assertEquals(ntlmDomain, proxyInfo.getNtlmDomain().orElse(null));
-        assertEquals(ntlmWorkstation, proxyInfo.getNtlmWorkstation().orElse(StringUtils.EMPTY));
+    private ProxyConfiguration createProxyConfiguration(String userName, String expectedPassWord) {
+        ProxyConfiguration proxyConfiguration = new ProxyConfiguration(expectedProxyHost, expectedProxyPort, userName, null, hostToExclude);
+
+        if (expectedPassWord == null || expectedPassWord.isEmpty()) {
+            return proxyConfiguration;
+        } else {
+            ProxyConfiguration spiedProxyConfiguration = Mockito.spy(proxyConfiguration);
```

Comment:
Why are we using Mockito.spy here when we could have passed the password in on line 126?

Reply:
I added a comment just now. This would involve low level Jenkins encryption in order to inject a password. Using anything but null causes a NPE. Eric and I actually worked on it for about an hour and couldn't get it to work with anything but null. That doesn't mean we can't get it to work, but would it be worth it to spend the time for this? We talked about the possibility of using JenkinsRule, which I had trouble using outside of a project that was a Jenkins plugin. I also question if we need to use a password, since we only need it to call the constructor with  `proxyConfiguration.getPassword()`.

What are your thoughts?

Reply:
It seems like the problematic path is when Secret::decrypt calls HistoricalSecrets::decrypt. One can work around this by surrounding the password in curly braces, but I think at that point we're muddying the test code to serve code we don't actually care about testing. This mockito.spy solution makes a lot more sense given that passing the password is nontrivial.